<div style="border:solid Chocolate 2px; padding: 40px">

<b>Ольга, привет!</b>

Меня зовут Евгений Головин, я буду ревьюером твоего проекта. Если ты не против, то предлагаю посмотрить наше общение на "ты" ;) Если удобнее на "вы", то нет проблем, только скажи об этом. 

В ходе работы я оставил тебе комментарии <font color='green'>зеленого</font>, <font color='gold'>желтого</font> и <font color='red'>красного</font> цветов. Сейчас объясню, что они значат:

<br/>

<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> Решение на этом шаге является полностью правильным.
</div>

<br/>

<div class="alert alert-warning">
    <h2> Комментарий ревьюера <a class="tocSkip"> </h2>
    
<b>Небольшие замечания и рекомендации💡:</b> Решение на этом шаге станет еще лучше, если внести небольшие коррективы.
</div>


<br/>
<div class="alert alert-block alert-danger">
<h2> Комментарий ревьюера <a class="tocSkip"></h2>

    
<b>На доработку🤔:</b>
 Решение на этом шаге требует существенной переработки и внесения правок. Напоминаю, что проект не может быть принят с первого раза, если ревью содержит комментарии, рекомендующие доработать шаги.
</div>
    
Увидев мой комментарий, не удаляй его, он будет очень полезен в случае повторной проверки работы :)
    
<div class="alert alert-info">
<b>А свой помечай вот так, чтобы я его не потерял ;)</b> 
</div>
    
Увидев мой комментарий, не удаляй его, он будет очень полезен в случае повторной проверки работы :)
    
На мои комменатрии можно и нужно реагировать, только делать это стоит так, чтобы твои и мои комменты не смешались: выделяй свои цветами, сильно отличающимися от моих.
    
Увидев у тебя неточность, в первый раз я лишь укажу на ее наличие и дам тебе возможность самому найти и исправить ее. На реальной работе твой руководитель будет поступать также, и я пытаюсь подготовить тебя именно к работе аналитиком. Но если ты пока не справишься с такой задачей - при следующей проверке я дам более точную подсказку!

# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> Тогда давай туда и переноситься!)

## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [2]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [3]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [5]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [6]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [7]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [8]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [9]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [10]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [11]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [12]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [13]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [14]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [15]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [16]:
data.duplicated().sum()

71

In [17]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [18]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [19]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [20]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [21]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [22]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> ВЖУХ и мы здесь!

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [24]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
data_kids = data.pivot_table(index = ['children'], values = 'debt', aggfunc = ['mean', 'sum', 'count'])
display(data_kids)

mean   sum  count
              debt  debt   debt
children                       
0         0.075438  1063  14091
1         0.092346   444   4808
2         0.094542   194   2052
3         0.081818    27    330
4         0.097561     4     41
5         0.000000     0      9

<div class="alert alert-warning">
    <h2> Комментарий ревьюера <a class="tocSkip"> </h2>
    
<b>Небольшие замечания и рекомендации💡:</b> Хороший код, но я бы рекомендовал использовать display() вместо print(), так будет читабельнее

**Вывод:** 
в таблице:
mean - процент не вернувших кредит в срок от общего числа семей с указанным количеством детей, где 
0 - вернули в срок, 1 - не вернули
sum - количество семей, не вернувших кредит
count - общее количество семей с указанным количеством детей

Из таблицы видно, что общее количество семей с разным количеством детей сильно разнится, семей без детей, взявших кредит, 14091, а семей с 5 детьми - всего 9. То есть выборки получаются не корректными для сравнения. Тем не менее, если не брать в расчет семьи с 5 детьми, как сильно выбивающиеся из общей картины, то можно увидеть, что процент семей без детей возвращающих кредит в срок, выше, чем у семей с детьми.

<div class="alert alert-warning">
    <h2> Комментарий ревьюера <a class="tocSkip"> </h2>
    
<b>Небольшие замечания и рекомендации💡:</b> Верно, но давай еще поработаем с деталями. Почему мы не берем в расчет заемщиков с 5 детьми? Потому что выборка малочисленная и вывод по ней будет не репрезентативен? Можно ли то же самое сказать про заемщиков с 3 и 4 детьми?

<div class="alert alert-info">
<b>Женя, спасибо за комментарий. Я действительно упустила из виду, что выборка для семей с 3 и 4 детьми тоже невелика. То есть их тоже можно отбросить. Наиболее релевантны здесь выборки с 1 и 2 детьми против семей без детей.</b> 
</div>

<div class="alert alert-success">
<h2> Комментарий ревьюера v2 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> Хорошо)

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [25]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
data_status = data.pivot_table(index = ['family_status'], values = 'debt', aggfunc = ['mean', 'sum', 'count'])
display(data_status)


mean  sum  count
                           debt debt   debt
family_status                              
Не женат / не замужем  0.097639  273   2796
в разводе              0.070648   84   1189
вдовец / вдова         0.066246   63    951
гражданский брак       0.093130  385   4134
женат / замужем        0.075606  927  12261

**Вывод:** 
в таблице:
mean - процент не вернувших кредит в срок от общего числа людей с указанным семейным положением, где 
0 - вернули в срок, 1 - не вернули
sum - количество людей, не вернувших кредит
count - общее количество людей с указанным семейным положением

В данной таблице также есть сильно выбивающая группа - это те, чье семейное положение "женат / замужем". Остальные группы по численности более менее похожи друг на друга. Те, у кого семейное положение "Не женат / не замужем" и "гражданский брак" не возвращают кредит в срок практически в 10% случаев, в отличие от тех, у кого семейное положение "в разводе" и "вдовец / вдова", среди которых процент не вернувших кредит в срок около 6-7%.

<div class="alert alert-warning">
    <h2> Комментарий ревьюера <a class="tocSkip"> </h2>
    
<b>Небольшие замечания и рекомендации💡:</b> Отлично, но давай искать тенденцию. Что можем сказать про заемщиков, когда-либо в брак вступавших?

<div class="alert alert-info">
<b>Я в целом указала, что группы тех, чье семейное положение "в разводе" и "вдовец / вдова", возвращают кредиты лучше. Тут стоит подчеркнуть обязательно, что это те, кто был в браке? </b> 
</div>

<div class="alert alert-success">
<h2> Комментарий ревьюера v2 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> В целом здесь выделяется группа из статусов в разводе, вдовец/вдова, женат/замужем, поэтому такой итог стоит подвести, да. Ведь мы же ищем тенденции

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [26]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
# Для сравнения возьмем столбец total_income_category, в котором в коде выше (2.6) разбивали уровень дохода по категориям
data_income = data.pivot_table(index = ['total_income_category'], values = 'debt', aggfunc = ['mean', 'sum', 'count'])
display(data_income)


mean   sum  count
                           debt  debt   debt
total_income_category                       
A                      0.080000     2     25
B                      0.070602   354   5014
C                      0.084982  1353  15921
D                      0.060172    21    349
E                      0.090909     2     22

**Вывод:** 
в таблице:
mean - процент не вернувших кредит в срок от общего числа людей с указанным уровнем дохода, где 
0 - вернули в срок, 1 - не вернули
sum - количество людей, не вернувших кредит
count - общее количество людей с указанным уровнем дохода

Из таблицы видно, что общее количество людей с разным уровнем дохода сильно разнится, в категориях А и Е очень маленькие выборки, всего по 25 и 22 человека, а категория С оказалась самой большой, в ней 15921 человек. Настолько разные выборки сравнивать некорректно.

<div class="alert alert-block alert-danger">
<h2> Комментарий ревьюера <a class="tocSkip"></h2>

<b>На доработку🤔:</b>  Не репрезентативные между собой точно нельзя сравшивать, а вот репрезентативные выборки можно. Поэтому давай исключать малеьнкие выборки и работать с тем, что останется

<div class="alert alert-info">
<b> Исключив маленькие выборки, оставляем для сравнения 2 категории - группа с доходом В - 200001–1000000 
и группа с доходом С - 50001–200000. В группе В процент не вернувших вовремя кредит составляет 7%, в то 
время как в группе С - 8,5%. Можем сделать вывод, что чем выше доход, тем выше вероятность возврата 
кредита в срок. 
P.S. Меня правда немного смущает, что выборки это разнятся на 10000 человек. Это всё равно считаем репрезентативной
выборкой?</b> 
</div>

<div class="alert alert-success">
<h2> Комментарий ревьюера v2 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> Вывод верный, выборки считать репрезентативными. Длярепрезентативности введен процентаж - категорияз должна состять хотя бы и 5% от суммарного количества строк в датафрейме. То, что категории отличаются по численности - факт, но вывод делать можно

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [29]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
# Для сравнения возьмем столбец purpose_category, в котором в коде выше (2.6) разбивали цели кредита по категориям
data_purpose = data.pivot_table(index = ['purpose_category'], values = 'debt', aggfunc = ['mean', 'sum', 'count'])
display(data_purpose)

mean  sum  count
                              debt debt   debt
purpose_category                              
операции с автомобилем    0.093480  400   4279
операции с недвижимостью  0.072551  780  10751
получение образования     0.092528  369   3988
проведение свадьбы        0.079118  183   2313

**Вывод:** 
в таблице:
mean - процент не вернувших кредит в срок от общего числа людей с указанной целью кредита, где 
0 - вернули в срок, 1 - не вернули
sum - количество людей, не вернувших кредит
count - общее количество людей с указанной целью кредита

В таблице есть сильно выбивающая по численности группа - это те, кто брал кредит на операции с недвижимостью. Остальные группы по численности более менее похожи друг на друга. Процент не вернувших вовремя кредит не сильно зависит от цели кредита, он колеблется примерно от 7,9% до 9,3%. 

<div class="alert alert-warning">
    <h2> Комментарий ревьюера <a class="tocSkip"> </h2>
    
<b>Небольшие замечания и рекомендации💡:</b> И все же банку важно понять, какие цели более надежны, какие менее надежны, поэтому я бы отметил это здесь

<div class="alert alert-info">
<b> Стоит отметить, что среди тех, кто брал кредит на операции с недвижимостью и на проведение свадьбы
процент не вернувших вовремя кредит, ниже, чем тех, кто брал кредиты на авто и образование - примерно 7% 
против 9%.</b> 
</div>

<div class="alert alert-success">
<h2> Комментарий ревьюера v2 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> Верно!

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* 
В исходных данных пропуски появились в стобцах days_employed и total_income. Это могли быть случайные пропуски, чтобы не ставить 0 или в силу человеческого фактора. Могли быть неслучайные пропуски, кто-то намеренно не хотел отвечать на эти вопросы. 

<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> Возможно, да. Также возможно, что произошла техническая ошибка - выгрузки данных, например

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 
Мы не можем увидеть, нет ли нестандартных значений в огромном массиве данных количественных переменных. Если таковые имеются, то смотреть среднее значение здесь будет уже неправильно, оно будет отклоняться в сторону нестандартных значений. Поэтому медиана при работе с количественными переменными лучше, она находится точно посередине всех значений. 

<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> Верно! Медиана нивелирует выбросы, а вот среднее прифметическое нет

### Шаг 4: общий вывод.

Напишите ваш общий вывод.

1. Семьи без детей отдают возвращают кредиты в сроки "активнее", чем семьи с детьми - процент 
должников среди семей без детей 7,5%, в то время как среди семей с детьми эта цифра около 9%.
2. Те, у кого семейное положение "Не женат / не замужем" и "гражданский брак" не возвращают 
кредит в срок практически в 10% случаев, в отличие от тех, у кого семейное положение "в разводе" и 
"вдовец / вдова", среди которых процент не вернувших кредит в срок около 6-7%.
3. Не удалось сравнить проценты должников по уровню дохода, так как выборки были слишком разные. 
4. Цели кредита не оказывают существенного влияния на возврат кредитов. Они примерно везде одинаковые -
от 7,9% до 9,3%.

<div class="alert alert-block alert-danger">
<h2> Комментарий ревьюера <a class="tocSkip"></h2>

<b>На доработку🤔:</b>  Поправми ошибки и вернемся сюда, хорошо?

<div class="alert alert-info">
<b><p> 1. Семьи без детей отдают возвращают кредиты в сроки "активнее", чем семьи с детьми - процент должников среди семей без детей 7,5%, в то время как среди семей с детьми эта цифра около 9%.</p>
<p>2. Те, у кого семейное положение "Не женат / не замужем" и "гражданский брак" не возвращают кредит в срок практически в 10% случаев, в отличие от людей, когда-либо вступавших в брак, среди которых процент не вернувших кредит в срок около 6-7%.</p>
<p>3. Чем выше доход, тем выше вероятность возврата кредита в срок.</p> 
<p>4. Цели кредита не оказывают существенного влияния на возврат кредитов. Тем не менее, те, кто берет кредиты на операции с недвижимостью и на свадьбу, возвращают
кредит в срок, чаще, чем те, кто берет их на операции с автомобилем и на образование.</p>
</b> 
</div>

<div class="alert alert-success">
<h2> Комментарий ревьюера v2 <a class="tocSkip"> </h2>

<b>Все супер!👍:</b> Хорошо

<div style="border:solid Chocolate 2px; padding: 40px">
    
**Общий вывод по проекту ревьювера**:
 Ольга, спасибо за  проект! Ты приложила много усилий, чтобы довести его до конца, проделана огромная работа, и это видно невооруженным глазом, ты большой молодец!
    
**Отмечу положительные моменты**:
    
    1. Все разложено по полочкам, всегда понятен ход твоих мыслей, приятно смотреть
 
    2. На разных шагах проекта ты предлагала очень интересные решения и методы для их реализации. Хорошая работа!
    
    
**На что стоит обратить внимание**:
    
    1. На численность категорий и возможность делать по ним выводы
    
    
**Удачи и жду твой проект на повторное ревью!**

<div style="border:solid Chocolate 2px; padding: 40px">
    
**Общий вывод по проекту ревьювера v2**:
 Ольга, спасибо за работу, проект зачтен, поздравляю тебя! Успешной учебы и еще увидимся в Практикуме! ;)